code/game/objects/items/storage/holsters.dm

```diff
@@ -258,7 +261,11 @@
 	icon = 'icons/obj/clothing/belts.dmi'
 	base_icon = "t19_holster"
 	flags_equip_slot = ITEM_SLOT_BELT
-	holsterable_allowed = list(/obj/item/weapon/gun/smg/standard_machinepistol)
+	holsterable_allowed = list(
+		/obj/item/weapon/gun/smg/standard_machinepistol,
+		/obj/item/weapon/gun/smg/standard_machinepistol/compact,
+		/obj/item/weapon/gun/smg/standard_machinepistol/vgrip,
```

Comment:
why do you need to add the childs here

Reply:
The way the check works, it doesn't istype so the icon won't properly update otherwise. This was intentional, due to certain items like shotguns having 'real' child items you wouldn't want to include, but I'll probably need to think about this in the future when I add options to have multiple different overlays depending on the special item inserted (i.e. pistol belts).